app/Http/Controllers/TrainingDash.php

```diff
@@ -567,67 +573,366 @@ public function otsCancel($id) {
+                ]
+            ]
+        );
+        $setmoreResp = (string) $res->getBody();
+        $setmoreRespA = json_decode($setmoreResp, true);
+        if (isset($setmoreRespA['data']['appointments'])) {
+            $setmoreAppointments = array_merge($setmoreAppointments, $setmoreRespA['data']['appointments']);
+        }
+        if (isset($setmoreRespA['data']['cursor'])) {
+            $cursor = $setmoreRespA['data']['cursor'];
+            return true;
+        }
+        return false;
+    }
+
+    public function generateGraphs(Request $request) {
+        $graphId = $request->id;
+        $statArr = TrainingDash::generateTrainingStats($request->year, $request->month, 'graph');
+        // Reformat date range
+        $from = Carbon::createFromDate($statArr['date']['start_date'])->format('m/j/Y');
+        $to = Carbon::createFromDate($statArr['date']['end_date'])->format('m/j/Y');
+        MtJpGraph::load(['bar', 'plotline']);
+        $graph = new \Graph(600, 600);
+        $graph->SetFrame(false, 'black', 0);
+        $graph->SetScale("textlin");
+        // Graph configuration
+        $statsGraphs = [
+            1 => ['title' => 'Sessions Per Month', 'subtitle' => '(' . $from . ' - ' . $to . ')', 'x-title' => 'Session Type', 'y-title' => 'Number of Sessions', 'dataset' => 'sessionsByType'],
+            2 => ['title' => 'Sessions Per Month By Instructor/Mentor', 'subtitle' => '(' . $from . ' - ' . $to . ')', 'x-title' => '', 'y-title' => 'Number of Sessions', 'dataset' => null],
+            3 => ['title' => 'Average Session Duration', 'subtitle' => 'Last Six Months', 'x-title' => '', 'y-title' => 'Average Session Duration (minutes)', 'dataset' => 'sessionDuration'],
+            4 => ['title' => 'Students Requiring Training', 'subtitle' => 'As of ' . Carbon::now()->format('m/d/Y'), 'x-title' => 'Student Type', 'y-title' => 'Number of Students', 'dataset' => 'studentsRequireTng']
+        ];
+        $graph->title->Set($statsGraphs[$graphId]['title']);
+        $graph->subtitle->Set($statsGraphs[$graphId]['subtitle']);
+        $graph->xaxis->SetTitle($statsGraphs[$graphId]['x-title'], 'center');
+        $graph->yaxis->SetTitle($statsGraphs[$graphId]['y-title'], 'middle');
+        $graph->title->SetFont(FF_FONT1, FS_BOLD);
+        $graph->yaxis->title->SetFont(FF_FONT1, FS_BOLD);
+        $graph->xaxis->title->SetFont(FF_FONT1, FS_BOLD);
+
+        if (in_array($graphId, [1, 4])) { // Sessions per month, students requiring training by category
+            $bplot = new \BarPlot(array_values($statArr[$statsGraphs[$graphId]['dataset']]));
+            $graph->Add($bplot);
+            $graph->xaxis->SetTickLabels(array_keys($statArr[$statsGraphs[$graphId]['dataset']]));
+        } elseif ($graphId == 2) { // Sessions by instructor per month
+            $instructors = $plotArray = [];
+            $instructionalCategories = ['S1', 'S2', 'S3', 'C1', 'Other'];
```

Comment:
Instead of writing comments for these types of graphs, it is more readable to create static global variables:

```php
public static final $GRAPH_TYPE_ONE = 1;
...
```

then to use:
```php
TrainingDash::$GRAPH_TYPE_ONE
```

Reply:
Great suggestion... I would have never thought of doing that. I'll incorporate this into the code